src/emulator/dataconnect/pgliteServer.ts

```diff
@@ -50,41 +58,79 @@ export class PostgresServer {
         server.emit("error", err);
       });
     });
+
     const listeningPromise = new Promise<void>((resolve) => {
       server.listen(port, host, () => {
         resolve();
       });
     });
-    await db.waitReady;
     await listeningPromise;
     return server;
   }
 
   async getDb(): Promise<PGlite> {
-    if (this.db) {
-      return this.db;
+    if (!this.db) {
+      // Not all schemas will need vector installed, but we don't have an good way
+      // to swap extensions after starting PGLite, so we always include it.
+      const vector = (await dynamicImport("@electric-sql/pglite/vector")).vector;
+      const uuidOssp = (await dynamicImport("@electric-sql/pglite/contrib/uuid_ossp")).uuid_ossp;
+      const pgliteArgs: PGliteOptions = {
+        username: this.username,
+        database: this.database,
+        debug: 0,
+        extensions: {
+          vector,
+          uuidOssp,
+        },
+        // TODO:  Use dataDir + loadDataDir to implement import/export.
+        dataDir: this.dataDirectory,
+        // loadDataDir?: Blob | File; // This will be used with .dumpDataDir() for import/export
+      };
+      if (this.importPath) {
+        logger.debug(`Importing from ${this.importPath}`);
+        const rf = fs.readFileSync(this.importPath);
+        const file = new File([rf.buffer], this.importPath);
+        pgliteArgs.loadDataDir = file;
+      }
+      this.db = await PGlite.create(pgliteArgs);
+      await this.db.waitReady;
     }
-    // Not all schemas will need vector installed, but we don't have an good way
-    // to swap extensions after starting PGLite, so we always include it.
-    const vector = (await dynamicImport("@electric-sql/pglite/vector")).vector;
-    const uuidOssp = (await dynamicImport("@electric-sql/pglite/contrib/uuid_ossp")).uuid_ossp;
-    return PGlite.create({
-      username: this.username,
-      database: this.database,
-      debug: 0,
-      extensions: {
-        vector,
-        uuidOssp,
-      },
-      // TODO:  Use dataDir + loadDataDir to implement import/export.
-      // dataDir?: string;
-      // loadDataDir?: Blob | File;
-    });
+    return this.db;
+  }
+
+  public async clearDb(): Promise<void> {
+    // if (this.dataDirectory) {
+    //   rmSync(this.dataDirectory, { recursive: true });
```

Comment:
What are these commented lines for?

Reply:
Sorry, accidentally sent this out for review earlier than intended - the implementation here has some minor discrepancies from what was approved and needs some clean up.

Will re-request once its in a better state